### PR TITLE
Feature/cmake config

### PR DESCRIPTION
**Pull request by kgerheiser**
Remove version from library name

Use CMake package config

PFUnit tests (didn't convert the existing unit tests to use pfunit yet)

Github Actions to build and test on commit/PR

### CODE REVIEW

File: cmake/PackageConfig.cmake.in [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@PACKAGE_INIT@
+
+#  * @PROJECT_NAME@::@PROJECT_NAME@ - library target
+
+# Include targets file.  This will create IMPORTED target @PROJECT_NAME@
+include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake")
+
```
Comment (aerorahul) [inline cmake/PackageConfig.cmake.in:7]: Add find_dependency for dependencies

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+set(lib_name ${PROJECT_NAME})
+set(module_dir ${CMAKE_CURRENT_BINARY_DIR}/include)
+
+add_library(${lib_name} STATIC ${fortran_src})
+add_library(${PROJECT_NAME}::${lib_name} ALIAS ${lib_name})
+
+set_target_properties(${lib_name} PROPERTIES Fortran_MODULE_DIRECTORY ${module_dir})
+if(OpenMP_Fortran_FOUND)
+  set_target_properties(${lib_name} PROPERTIES COMPILE_FLAGS "${OpenMP_Fortran_FLAGS}")
```
Comment (aerorahul) [inline src/CMakeLists.txt]: use OpenMP::OpenMP_Fortran

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,87 +1,41 @@
...
-file(GLOB fortran_src
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/*.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/*.f90
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/*.F)
+if(NOT CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel|GNU|Clang|AppleClang)$")
+  message(
+    WARNING "Compiler not officially supported: ${CMAKE_Fortran_COMPILER_ID}")
+endif()
 
 set(CMAKE_FIND_PACKAGE_PREFER_CONFIG true)
```
Comment (aerorahul) [inline CMakeLists.txt]: please remove this line.
`set(CMAKE_FIND_PACKAGE_PREFER_CONFIG true)`

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,87 +1,41 @@
...
 endif()
 
-if(NOT TARGET ip_d)
-  find_package(ip REQUIRED)
-endif()
+add_subdirectory(src)
 
-if(NOT TARGET sp_d)
-  find_package(sp REQUIRED)
+if(ENABLE_TESTS)
```
Comment (aerorahul) [inline CMakeLists.txt:32]: I didn't see the `ENABLE_TESTS` option anywhere.

File: cmake/PackageConfig.cmake.in (hunk 1 above)
Comment (aerorahul) [reply cmake/PackageConfig.cmake.in:7]: Including the OpenMP option.  See NCEPLIBS-ip::develop for example.

**Review by aerorahul — CHANGES_REQUESTED** [review]
some changes requested on dependency handling.

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+set_target_properties(${lib_name} PROPERTIES INTERFACE_LINK_LIBRARIES ${lib_name})
+
+target_include_directories(${lib_name} PUBLIC
+  $<BUILD_INTERFACE:${module_dir}>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>)
+
+target_link_libraries(${lib_name} PRIVATE
+  nemsio::nemsio bacio::bacio_4 w3nco::w3nco_d ip::ip_d sp::sp_d)
+list(APPEND LIB_TARGETS ${lib_name})
+
```
Comment (aerorahul) [inline src/CMakeLists.txt:28]: Missing
```
install(DIRECTORY ${module_dir} DESTINATION ${CMAKE_INSTALL_PREFIX})
```

**Comment by kgerheiser** [comment]
Good @aerorahul?

**Comment by aerorahul** [comment]
> Good @aerorahul?

not yet.  The dependencies in `PackageConfig.cmake.in` need to be added in .

**Review by aerorahul — APPROVED** [review]
looks good now.

**Comment by DusanJovic-NOAA** [comment]
See https://github.com/NOAA-EMC/NCEPLIBS-nemsiogfs/issues/9. This library depends (directly) only on nemsio.

**Comment by Hang-Lei-NOAA** [comment]
Dusan is correct.
nemsio is the only dependency for this lib.

**Comment by kgerheiser** [comment]
Fixed

**Review by aerorahul — APPROVED** [review]
thanks for the fixes.

**Pull request merged**